src/basilisp/lang/runtime.py

```diff
@@ -1246,17 +1246,17 @@ def _to_py_kw(o: kw.Keyword, keyword_fn: Callable[[kw.Keyword], Any] = _kw_name)
     return keyword_fn(o)
 
 
-@to_py.register(llist.List)
+@to_py.register(IPersistentList)
```

Comment:
In #460 I discovered that `functools.singledispatch` can dispatch on interfaces (using an object's MRO) and it caches the final result, so this is more generic and just as fast as registering the object type directly after the first call.

Reply:
Fixed in #552